### PR TITLE
fix: nextjs sdk proxy configuration

### PR DESCRIPTION
**Pull request by Rutik7066**
## Summary

I added a check. If  `configureHighlightProxy` is `true` then `backendUrl` will set to  `/highlight-events`  and if `/highlight-events`  
 is set in `next.config,` it will not push the default rewrite.

/claim #9042

### CODE REVIEW

**Review by Vadman97 — CHANGES_REQUESTED** [review]
Hi @Rutik7066 , thanks for the PR. However this is not configuring the [HighlightInit](https://github.com/highlight/highlight/blob/main/sdk/highlight-next/src/next-client.tsx) component as described in the ticket.

**Comment by Rutik7066** [comment]
@Vadman97 I think I have misunderstood the issue. Can you please explain me in details what exactly we want. I noticed that when we add `/highlight-events` as backendUrl prop it will add localhost as host in Url is this we want to do it in function

**Comment by Rutik7066** [comment]
@Vadman97 

> @Vadman97 I think I have misunderstood the issue. Can you please explain me in details what exactly we want. I noticed that when we add `/highlight-events` as backendUrl prop it will add localhost as host in Url is this we want to do it in function

**Comment by Rutik7066** [comment]
@Vadman97 wait for your review.

**Comment by Rutik7066** [comment]
@Vadman97 if anything needs to be changed please let me know

File: sdk/highlight-next/src/util/with-highlight-config.ts [hunk 1]
```diff
@@ -117,6 +117,13 @@
 ) => {
 	const defaultOpts = await getDefaultOpts(config, highlightOpts)
 
+	if (defaultOpts.configureHighlightProxy ){
+			const backendUrl = process.env.NEXT_PUBLIC_HIGHLIGHT_BACKEND_URL;
+			const url = new URL(backendUrl || 'http://localhost:3000');
```
[the remark this answers is not in the document]
Comment (Rutik7066) [reply sdk/highlight-next/src/util/with-highlight-config.ts:122]: @Vadman97 we use locahost:3000 for our backend.

**Comment by Rutik7066** [comment]
@Vadman97  github action is failling because it is depreciated.
Screenshot from 2024-12-05 16-41-11

File: git
Comment (Vadman97) [inline git:1]: please remove this

**Review by Vadman97 — CHANGES_REQUESTED** [review]
this is not the right approach. we need to update https://github.com/highlight/highlight/blob/2339697c4ef5d3546e1ae2d2549dd417f2406657/sdk/highlight-next/src/next-client.tsx#L11-L38 to update `backendUrl` when the `/highlight-events` route is set up

**Comment by Rutik7066** [comment]
@Vadman97 While rebasing the PR, I accidentally closed it. Now, GitHub does not show an option to reopen it, so I will open a new PR.

**Pull request closed without merging**